File: core/src/main/java/com/timepath/vfs/MutableVFile.java
package com.timepath.vfs;

import org.jetbrains.annotations.NotNull;

/**
 * @param <V> Specific type of VFile
 * @author TimePath
 */
public interface MutableVFile<V extends VFile<?>> extends VFile<V> {

    @NotNull
    V add(@NotNull V file);

    @NotNull
    V addAll(@NotNull Iterable<? extends V> files);

    void remove(@NotNull V file);

    void removeAll(@NotNull Iterable<? extends V> files);
}

File: core/src/main/kotlin/com/timepath/vfs/MutableVFile.kt
package com.timepath.vfs

/**
 * @param <V> Specific type of VFile
 * @author TimePath
 */
public trait MutableVFile<V : VFile<V>> : VFile<V> {

    public fun add(file: V): V

    public fun addAll(files: Iterable<V>): V

    public fun remove(file: V)

    public fun removeAll(files: Iterable<V>)
}
